Update util_json

SalahElHazzat opened the pull request:
Added functions to manipulate and normalize jsons in dataset/json.

Entry point main will launch testing code that creates a Json_csv.csv then normalize jsons in dataset/json by creating a new repository dataset/normalized_jsons that contain same hierarchy but json structure is different.

pep8speaks commented:
Hello @SalahElHazzat! Thanks for opening this PR. 

* In the file [`mlmodels/util_json.py`](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py):

> [Line 186:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L186): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 187:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L187): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 187:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L187): [F811](https://duckduckgo.com/?q=pep8%20F811) redefinition of unused 'os' from line 7
> [Line 188:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L188): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 189:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L189): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 196:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L196): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 198:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L198): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 0
> [Line 199:48](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L199): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 200:35](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L200): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 207:79](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L207): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 217:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L217): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (3)
> [Line 224:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L224): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 0
> [Line 225:9](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L225): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 226:12](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L226): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 230:18](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L230): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 231:26](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L231): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 232:26](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L232): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 234:9](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L234): [E722](https://duckduckgo.com/?q=pep8%20E722) do not use bare 'except'
> [Line 235:23](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L235): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 237:20](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L237): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 238:23](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L238): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 240:34](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L240): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 245:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L245): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (3)
> [Line 248:37](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L248): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 249:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L249): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 251:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L251): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 0
> [Line 252:10](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L252): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 259:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L259): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (3)
> [Line 264:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L264): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 0
> [Line 266:10](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L266): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 267:14](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L267): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 271:17](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L271): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 273:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L273): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 275:36](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L275): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 275:42](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L275): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 275:54](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L275): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ':'
> [Line 280:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L280): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1
> [Line 282:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L282): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 283:7](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L283): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 284:6](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L284): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 285:13](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L285): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 287:9](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L287): [E741](https://duckduckgo.com/?q=pep8%20E741) ambiguous variable name 'l'
> [Line 287:10](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L287): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 288:10](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L288): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 288:24](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L288): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 288:26](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L288): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 291:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L291): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1
> [Line 292:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L292): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 0
> [Line 292:28](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L292): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 292:33](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L292): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 293:24](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L293): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 294:11](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L294): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 295:10](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L295): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 297:20](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L297): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 298:16](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L298): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 298:31](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L298): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 298:39](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L298): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 300:10](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L300): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 301:16](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L301): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 305:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L305): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1
> [Line 306:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L306): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 0
> [Line 307:10](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L307): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 308:7](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L308): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 309:13](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L309): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 310:13](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L310): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 312:11](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L312): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 313:17](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L313): [E271](https://duckduckgo.com/?q=pep8%20E271) multiple spaces after keyword
> [Line 314:18](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L314): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 315:19](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L315): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 316:41](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L316): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 316:44](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L316): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 318:10](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L318): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 319:19](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L319): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 320:12](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L320): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 321:21](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L321): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 323:11](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L323): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 324:13](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L324): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 325:19](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L325): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 327:25](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L327): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 328:24](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L328): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 329:19](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L329): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 329:27](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L329): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 330:25](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L330): [E271](https://duckduckgo.com/?q=pep8%20E271) multiple spaces after keyword
> [Line 330:36](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L330): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 330:38](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L330): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 330:46](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L330): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 331:25](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L331): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 331:30](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L331): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 331:35](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L331): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 333:20](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L333): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 338:40](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L338): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 340:18](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L340): [E211](https://duckduckgo.com/?q=pep8%20E211) whitespace before '('
> [Line 340:59](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L340): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 341:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L341): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 343:39](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L343): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 343:57](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L343): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 343:62](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L343): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 343:74](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L343): [E226](https://duckduckgo.com/?q=pep8%20E226) missing whitespace around arithmetic operator
> [Line 344:43](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L344): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 344:45](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L344): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 348:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L348): [E266](https://duckduckgo.com/?q=pep8%20E266) too many leading '#' for block comment
> [Line 349:1](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L349): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1
> [Line 350:15](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L350): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 351:25](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L351): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 352:17](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L352): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 353:7](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L353): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 357:14](https://github.com/arita37/mlmodels/blob/ead57362ec8651003006deb0be7edea3ca9cba2f/mlmodels/util_json.py#L357): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator

Do see the [Hitchhiker's guide to code style](https://goo.gl/hqbW4r)

arita37 commented on mlmodels/util_json.py:
```diff
@@ -183,4 +183,175 @@ def test_functions_json(arg=None):
+    Please change path by using os.chdir(***json directory Path***)
+    
+'''
+def getListOfJsonsPaths(dirName):
```
Follow my naming convention :

json_xxx_yyyy(
os_folder_getall(

arita37 commented on mlmodels/util_json.py:
```diff
@@ -183,4 +183,175 @@ def test_functions_json(arg=None):
+    'Path' key, containing path to the json and 'json' key containing the dictionary
+
+'''
+def Jsons_ToDictionaries(json_Paths):
```
No upper case

json_todict()

arita37 commented on mlmodels/util_json.py:
```diff
@@ -183,4 +183,175 @@ def test_functions_json(arg=None):
+
+
+
+'''
```
Google
  Docs Strings in python

Or use pycharm editor

arita37 commented on mlmodels/util_json.py:
```diff
@@ -183,4 +183,175 @@ def test_functions_json(arg=None):
+    Returns the list of dictionaries 
+    
+'''
+def IndexedDicts_ToDicts(indexed_Dicts):
```
Put it as internal function
no need to pollute with too small function

arita37 commented on mlmodels/util_json.py:
```diff
@@ -183,4 +183,175 @@ def test_functions_json(arg=None):
+    return Dict
+
+'''transform csv to list of dicts and create new normalized jsons'''
+def csv_toJsons(csv):
```
csv_to_json

arita37 commented on mlmodels/util_json.py:
```diff
@@ -183,4 +183,175 @@ def test_functions_json(arg=None):
+    Indexed_Dictionaries=Jsons_ToDictionaries(JsonsPaths)
+    Dictionaries=IndexedDicts_ToDicts(Indexed_Dictionaries)
+    Df=IndexedDicts_ToDF(Indexed_Dictionaries)
+    Df.to_csv('Jsons_Csv.csv')
```
use

path_norm( “ztmp/myfile”)

ztmp is gitignore
path_norm is mlmodels

SalahElHazzat commented:
Hi,
I used Pycharm to edit my code and followed your instructions:
- changed functions names as you recommended
- added docstrings to comment my functions
- used path_norm in code execution block

Please let me know if there is any further modifications you want to do.

arita37 commented on mlmodels/util_json.py:
```diff
@@ -135,52 +138,236 @@ def load_callable_from_uri(uri):
+import time
+
+
+def os_folder_getall(folder):
```
os_folder_getfiles

arita37 commented on mlmodels/util_json.py:
```diff
@@ -135,52 +138,236 @@ def load_callable_from_uri(uri):
+
+# Testing code
+if __name__ == "__main__":
+    json_path = path_norm("dataset\\json")
```
create    test_json_conversion()

arita37 commented on mlmodels/util_json.py:
```diff
@@ -135,52 +138,236 @@ def load_callable_from_uri(uri):
+    print("Now creating normalized jsons directory")
+    for p in paths_tocreate1:
+        try:
+            os.mkdir('normalized_jsons' + p)
```
os makedirs exist ok=true

arita37 commented on mlmodels/util_json.py:
```diff
@@ -135,52 +138,236 @@ def load_callable_from_uri(uri):
+    return d
+
+
+def csv_to_skeleton(csv):
```
Include this in other function

arita37 commented on mlmodels/util_json.py:
```diff
@@ -135,52 +138,236 @@ def load_callable_from_uri(uri):
+    return data
+
+
+def dicts_to_df(indexed_dicts):
```
Merge all in one function 
root path —-> output : dataframe all merged json.

SalahElHazzat commented:
Hi,
I made modifications you asked for:
- renamed os_folder_getall(folder) to test_json_conversion(folder)
- created test_json_conversion()
- included usage of os.makedirs() and os.path.exists()
- merged functions as you asked for

**Question:** 
I found that using path_norm( “ztmp/myfile”) is useless in my code, so I removed it.
Can you please launch on your machine to test if code runs correctly?

arita37 commented on mlmodels/util_json.py:
```diff
@@ -159,28 +159,183 @@ def test_functions_json(arg=None):
+        dicts.append(dd)
+
+    paths = list(ddf['file_path'])
+    cur_dir=os.getcwd()
```
Please use
from mlmodels.util import os_package_root_path
to get current root path

SalahElHazzat commented:
I updated code to use  os_package_root_path( ) function.

Hope code runs smoothly !

SalahElHazzat commented:
Hi I added usage of path_norm ( ) in the last commit !
